snap/hooks/install

```diff
@@ -8,14 +16,24 @@ export CONF="${SNAP_DATA}"/etc/$COMPONENT
 export LOGS="${SNAP_COMMON}"/var/log/$COMPONENT
 export DATA="${SNAP_COMMON}"/var/lib/$COMPONENT
 
+if [stat -c '%u' "$DATA" == 0 -o ! -d "$DATA"]; then
+    mkdir -p $DATA
+    mkdir -p $LOGS
+
+    chmod -R 770 "${SNAP_COMMON}"
+    chown -R 584788:root "${SNAP_COMMON}"
+    chmod 770 "${DATA}"
```

Comment:
**question** I know we have been using `770` before. I'm just wondering whether for the `data` folder (where Kafka, ZooKeeper) writes, maybe we could lower the permission for the group to read only (i.e. `750`).

Reply:
Testing changes on snap is a little tricky, as I do not have permissions to publish branches. I'll keep this PR opened while I work on the Kafka snap and test there

Reply:
If this is not blocking other things, I'd give it a go. I suppose I should also work to enable you guys to publish temp branches for testing. Let me double check with the other managers

Reply:
Alright, the CI is green with 750!
https://github.com/canonical/zookeeper-operator/actions/runs/8796703126